src/main/java/org/openstreetmap/atlas/checks/validation/tag/InvalidAccessTagCheck.java

```diff
@@ -0,0 +1,164 @@
+                "minimum.highway.type", MINIMUM_HIGHWAY_TYPE_DEFAULT);
+        this.minimumHighwayType = Enum.valueOf(HighwayTag.class, highwayType.toUpperCase());
+    }
+
+    /**
+     * This function will validate if the supplied atlas object is valid for the check. Objects
+     * passed to this function have already been filtered by the tags.filter parameter in the
+     * configuration file.
+     *
+     * @param object
+     *            the atlas object supplied by the Atlas-Checks framework for evaluation
+     * @return {@code true} if this object should be checked
+     */
+    @Override
+    public boolean validCheckForObject(final AtlasObject object)
+    {
+        return AccessTag.isNo(object) && ((object instanceof Edge) || (object instanceof Line))
+                && Edge.isMasterEdgeIdentifier(object.getIdentifier())
+                && !this.isFlagged(object.getOsmIdentifier()) && isMinimumHighway(object);
+    }
+
+    /**
+     * This is the actual function that will check to see whether the object needs to be flagged.
+     *
+     * @param object
+     *            the atlas object supplied by the Atlas-Checks framework for evaluation
+     * @return an optional {@link CheckFlag} object that
+     */
+    @Override
+    protected Optional<CheckFlag> flag(final AtlasObject object)
+    {
+        if (!isInMilitaryArea((LineItem) object))
+        {
+            this.markAsFlagged(object.getOsmIdentifier());
+            return Optional.of(this.createFlag(object,
+                    this.getLocalizedInstruction(0, object.getOsmIdentifier())));
+        }
+        return Optional.empty();
+    }
+
+    /**
+     * Checks if {@link LineItem} is inside an {@link Area} or {@link Relation} with tag
+     * {@code landuse=MILITARY} or tag key {@code military}.
+     *
+     * @param object
+     *            {@link LineItem} to check
+     * @return {@code true} if input {@link LineItem} is in an {@link Area} or {@link Relation} with
+     *         tag {@code landuse=MILITARY} or tag key {@code military}
+     */
+    private boolean isInMilitaryArea(final LineItem object)
+    {
+        for (final Area area : object.getAtlas()
+                .areas(area -> Validators.isOfType(area, LandUseTag.class, LandUseTag.MILITARY)
+                        || Validators.hasValuesFor(area, MilitaryTag.class)))
+        {
+            final Polygon areaPolygon = area.asPolygon();
+            if (object.intersects(areaPolygon)
+                    || areaPolygon.fullyGeometricallyEncloses(object.asPolyLine()))
+            {
+                return true;
+            }
+        }
+        for (final Relation relation : object.getAtlas().relations(
+                relation -> (Validators.isOfType(relation, LandUseTag.class, LandUseTag.MILITARY)
+                        || Validators.hasValuesFor(relation, MilitaryTag.class))
+                        && relation.isMultiPolygon()))
```

Comment:
Again this method is iterating over all the relations.

Reply:
This, unfortunately, does not seem to have as easy a fix as with areas. There does not appear to be an equivalent to `areasCovering` for relations. The only thing I could find was `relationsWithEntitiesIntersecting`. Using this would create false positive flags for anything that was completely contained in a relation, if I am not mistaken. I do not see an alternative to iterating over all relations, but would love to consider anything you might be able to come up with. I only need to test multipolygon relations for the intersection or containment of the current check object.

Reply:
See my other comment please. Retrieve relations intersecting line bounds and then filter them down.

Reply:
Thanks for the guidance. Your solution for areas is great. Unfortunately, it does not work in the case of relations. Using `relationsWithEntitiesIntersecting` will create false positive flags for items with bounds that are completely contained by the relation. 

Example:
https://www.openstreetmap.org/way/251110595 contained by
https://www.openstreetmap.org/relation/7437523

Reply:
Iterating over each and every relation should be avoided.

You can further filter relations returned from `relationsWithEntitiesIntersecting` method. That method accepts a second optional parameter: `Predicate<Relation>`. Something like:
```
object.getAtlas().relationsWithEntitiesIntersecting(object.bounds(), relation -> (Validators.isOfType(relation, LandUseTag.class, LandUseTag.MILITARY) || Validators.hasValuesFor(relation, MilitaryTag.class) && relation.isMultiPolygon()))
```
should give multi polygon relations with intersecting entities. Then `MultiPolygon` could be created from relation and checked for intersection.

Reply:
I agree that this should be avoided, but after all this I do not see an alternative. I thank you very much for all your suggestions, unfortunately, the use of `relationsWithEntitiesIntersecting` will create false positive flags. The LineItems in question do not intersect any of the members of the military relations that contain them, even when the bounds of the LineItems are used. Thus, no relations are passed to the predicate for further filtering.  

To completely substitute this loop, there would need to be a method that checks for containment of an item in a relation, not intersection. 

As another example:
Unit test `accessNoInHighwayEdgesLanduseMilitaryRelation` simulates this scenario.

Reply:
Sure, but that sounds like a problem in atlas - possibly missing an API that could check for intersection against the relation altogether. Feel free to create an issue on atlas project, document it here and then move forward. False positives are acceptable. However, iterating over all relations should be avoided given that the number of relations in an Atlas could be huge and such iteration will be repeated potentially for each Edge in the same Atlas.

Reply:
Alternatively, you can edit this check to go through military areas rather than `Edge`s and `Line`s. That way you can iterate military areas first and mark `Edge`s and `Line`s.

Reply:
Missing atlas relation method issue: [Relation Covering Method #137](https://github.com/osmlab/atlas/issues/137)